File: web/src/lib/money.ts
/** Format số tiền VND */
export function formatMoney(amount: number): string {
  return new Intl.NumberFormat('vi-VN', {
    style: 'currency',
    currency: 'VND',
    maximumFractionDigits: 0,
  }).format(amount);
}

/** Parse input tiền từ chuỗi (bỏ dấu chấm/phẩy) */
export function parseMoneyInput(value: string): number {
  const cleaned = value.replace(/[^\d]/g, '');
  return cleaned ? parseInt(cleaned, 10) : 0;
}

export function formatMoneyInput(amount: number): string {
  if (!amount) return '';
  return new Intl.NumberFormat('vi-VN').format(amount);
}

File: app/lib/utils/money.dart
import 'package:intl/intl.dart';

String formatMoney(num amount) {
  final s = NumberFormat('#,###', 'vi_VN').format(amount.round());
  return '${s.replaceAll(',', '.')}đ';
}

String formatMoneyInput(num amount) {
  if (amount == 0) return '';
  return NumberFormat('#,###', 'vi_VN').format(amount.round()).replaceAll(',', '.');
}

double parseMoneyInput(String raw) {
  final digits = raw.replaceAll(RegExp(r'[^\d]'), '');
  if (digits.isEmpty) return 0;
  return double.tryParse(digits) ?? 0;
}
